### ISSUE TITLE
Support problem matching for concurrent prefixed output, e.g. concurrently, npm-run-all, etc.

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#6655]
When working on monorepos it is common to need to run multiple watch/build commands in parallel using tools such as `pnpm -r`, `concurrently`, `npm-run-all`, `nx`, `turborepo`. Sometimes even in nested invocations, such as `pnpm --filter` from `npm-run-all`. Such tools often prefix each command's output which prevents using problem matchers with such setups.

Not only is the parallel output likely to confuse problem matching as it might match the output of parallel commands and pair them incorrectly (Watch begin/end patterns), but it will likely won't match at all as many/most problem matchers anchor their match to the start of the line for accuracy.

It would be nice to have some way to still use problem matchers in this scenario as they are quite convenient, being able to see at a glance if a background watch task errored and jumping to the error.
